src/wallet/rpcwallet.cpp

```diff
@@ -3668,7 +3668,8 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
     }
     if (meta) {
         ret.pushKV("timestamp", meta->nCreateTime);
-        if (!meta->hdKeypath.empty()) {
+        KeyOriginInfo info;
+        if (!meta->hdKeypath.empty() && pwallet->GetKeyOrigin(key_id, info)) {
```

Comment:
Would be nice to have a test that fails without this change.

Reply:
That may be a better idea and just close this PR